setup.py

```diff
@@ -26,7 +26,15 @@
 def build_envd_if_not_found():
     if not os.path.exists("bin/envd"):
         logging.info("envd not found. Build from scratch")
-        errno = subprocess.call(["make", "build-release"])
+        try:
+            with open(".GIT_TAG_INFO") as f:
```

Comment:
Where do we write the file? Is it maintained by us?

Reply:
https://github.com/tensorchord/envd/pull/1024/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R113
Seems like here. Newly added.

Reply:
However, I didn't get the whole logic. Can you explain the design and execution order? Thanks! @VoVAllen

Reply:
On macos arm, it built from source without git info. But the whole logic related to version depends on git tag. Missing git tag results in the bug. Thus here we manually save git tag info here

Reply:
nit: It's better to add comments here to explain the necessity of the code block. If one day macOS pre-built binary is released, this can be removed by consulting to the comment here.